geniza/corpus/solr_queryset.py

```diff
@@ -256,8 +259,17 @@ def get_highlighting(self):
         """highlight snippets within transcription/translation html may result in
         invalid tags that will render strangely; clean up the html before returning"""
         highlights = super().get_highlighting()
+        is_exact_search = "hl_query" in self.raw_params
         for doc in highlights.keys():
-            if "transcription" in highlights[doc]:
+            # _nostem fields should take precedence over stemmed fields in the case of an
+            # exact search; in that case, replace highlights for stemmed fields with nostem
+            if is_exact_search and "description_nostem" in highlights[doc]:
+                highlights[doc]["description"] = highlights[doc]["description_nostem"]
+            if is_exact_search and "transcription_nostem" in highlights[doc]:
+                highlights[doc]["transcription"] = [
+                    clean_html(s) for s in highlights[doc]["transcription_nostem"]
+                ]
```

Comment:
I had been thinking that you would include all the highlights in the context and then the template could display them in priority order.  I think it probably doesn't matter, maybe it's a little cleaner to do here.